internal/app/networkd/pkg/reg/reg.go

```diff
@@ -155,16 +155,39 @@ func toCIDR(family uint8, prefix net.IP, prefixLen int) string {
 }
 
 // Check implements the Health api and provides visibilty into the state of networkd.
-// Ready signifies the daemon (api) is healthy and ready to serve requests.
+// We determine networkd health based on neighbor (arp table) entries.
+// Under normal operating circumstances, there should be a steady state of reachable neighbor entries.
+// If we get in a situation where all of our neighbor entries are stale, then we're in trouble.
 func (r *Registrator) Check(ctx context.Context, in *empty.Empty) (reply *healthapi.HealthCheckResponse, err error) {
+	// Set initial state to unknown
 	reply = &healthapi.HealthCheckResponse{
 		Messages: []*healthapi.HealthCheck{
 			{
-				Status: healthapi.HealthCheck_SERVING,
+				Status: healthapi.HealthCheck_UNKNOWN,
 			},
 		},
 	}
 
+	var neighbors []rtnetlink.NeighMessage
+
+	neighbors, err = r.Conn.Neigh.List()
+	if err != nil {
+		return reply, err
+	}
+
+	// After getting a list of neighbors we can upgrade to not serving
+	reply.Messages[0].Status = healthapi.HealthCheck_NOT_SERVING
+
+	// Find at least one neighbor in a reachable state
+	for _, neighbor := range neighbors {
+		// State = 2 == reachable
+		// State = 4 == stale
+		if neighbor.State == 2 {
```

Comment:
Wonder if we should include `neighbor.State == 2 || neighbor.State == 4`. I think the transition time between the two states is ~5-10s of inactivity, so looking for reachable only might be too tight.

Reply:
can we also use some constants for these so that is more readable?

Reply:
yeah, couldn't find them last night but looks like these map up:
https://github.com/golang/sys/blob/63cb32ae39b28d6bb8e7e215c1fc39dd80dcdb02/unix/linux/types.go#L593-L601

Reply:
Would it be smart to test that the interface index matches interfaces managed by networkd?